core/src/main/java/io/fabric8/maven/core/util/validator/ResourceValidator.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.util.validator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.ValidationMessage;
+import io.fabric8.maven.core.util.ResourceClassifier;
+import io.fabric8.maven.core.util.validator.IgnoreRule;
+import io.fabric8.maven.docker.util.Logger;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Path;
+import javax.validation.metadata.ConstraintDescriptor;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Validates Kubernetes/OpenShift resource descriptors using JSON schema validation method.
+ * For Openshift it adds some some exceptions from JSON schema constraints and ignores some validation errors.
+ */
+
+public class ResourceValidator {
+
+    // TODO: Locate URL under fabric8 repositories
+    private final static String jsonSchemaPath = "https://raw.githubusercontent.com/garethr/%s-json-schema/master/%s-standalone/%s.json";
```

Comment:
It would be good to host versions in the fabric8 repos. I think we can check with the author of that repo if it's possible to have a package including e.g. all the *local* schemas in a maven artifact, so it can be included directly in the plugin, avoiding the dependency on the github repo. The local versions should not be so heavy.

Reply:
nice thought!